### PR TITLE
Delete users who are deleted

### PR DESCRIPTION
**Pull request by sethvargo**
**Release Note**

```release-note
Delete users who are deleted
```

/assign @mikehelmick

### CODE REVIEW

File: pkg/database/user.go [hunk 1]
```diff
@@ -418,7 +418,7 @@ func (db *Database) DeleteUser(u *User, actor Auditable) error {
 		}
 
 		// Delete the user
-		if err := tx.Delete(u).Error; err != nil {
+		if err := tx.Unscoped().Delete(u).Error; err != nil {
```
Comment (mikehelmick) [inline pkg/database/user.go:421]: question - does this impact auditing?

I want to make sure that someone's trail of codes issued or audit logs things doesn't immediately disappear.

originally this was "disable user"

**Review by mikehelmick — CHANGES_REQUESTED** [review]

File: pkg/database/user.go (hunk 1 above)
Comment (sethvargo) [reply pkg/database/user.go:421]: No, audit logs are intentionally not tied back to the user via a FK or database-level link

Comment (sethvargo) [reply pkg/database/user.go:421]: It was disable user, but then the cleanup job _used_ to delete them. I think Brendan got caught in a weird place where he had performed the old action of "disable", but then the cleanup job was updated to focus on memberships and never deleted him.

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**